src/qililab/waveforms/flat_top.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Square waveform."""
+import numpy as np
+from scipy.special import erf
+
+from qililab.qprogram.decorators import requires_domain
+from qililab.qprogram.variable import Domain
+
+from .waveform import Waveform
+
+
+class FlatTop(Waveform):  # pylint: disable=too-few-public-methods
+    """Flat top Gaussian rise waveform.
+
+    Args:
+        duration (int): Duration of the pulse (ns).
+        amplitude (float): Maximum amplitude of the pulse.
+        gaussian (float, optional): Sigma number of the gaussian pulse shape. Defaults to 0.5.
+        buffer (float, optional): Buffer of the waveform. Defaults to 3.0.
+    """
+
+    @requires_domain("amplitude", Domain.Voltage)
+    @requires_domain("duration", Domain.Time)
+    def __init__(self, amplitude: float, duration: int, gaussian: float = 0.5, buffer: float = 3.0):
```

Comment:
add `requires_domain(Domain.Scalar)` for `gaussian` and `buffer` parameters. Also is the `gaussian` the same as `num_sigmas` parameter in the `Gaussian` waveform? If yes, maybe we should rename to be consistent?

Reply:
Also, why did we choose these default values? Is it a special case?

Reply:
yes, these were set by @paulqili

Reply:
`gaussian` is not the same as `num_sigmas` though it is similar. According to Paul it regulates how "gaussian" the flat top distribution is. But since it has a different nature I'd keep the name as is.